core/templates/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -437,23 +437,16 @@ export class InteractionObjectFactory {
     }
   }
 
-  createFromBackendDict(
-      interactionDict: InteractionBackendDict): Interaction {
-    var defaultOutcome;
-    if (interactionDict.default_outcome) {
-      defaultOutcome = this.outcomeFactory.createFromBackendDict(
-        interactionDict.default_outcome);
-    } else {
-      defaultOutcome = null;
-    }
-
+  createFromBackendDict(interactionDict: InteractionBackendDict): Interaction {
     return new Interaction(
       this.generateAnswerGroupsFromBackend(interactionDict.answer_groups),
       interactionDict.confirmed_unclassified_answers,
       this.convertFromCustomizationArgsBackendDict(
         interactionDict.id,
         interactionDict.customization_args),
-      defaultOutcome,
+      interactionDict.default_outcome ? (
+        this.generateOutcomeFromBackend(
```

Comment:
This is named wrongly. It should be generateOutcomeFromBackendDict

Reply:
OK, I've also changed the other incorrect names in the file.